src/app/examples/todos/todos.component.spec.ts

```diff
@@ -4,18 +4,44 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { CoreModule } from '@app/core';
 import { SharedModule } from '@app/shared';
 
-import { ExamplesModule } from '../examples.module';
-
 import { TodosComponent } from './todos.component';
+import { Store, StoreModule } from '@ngrx/store';
+import {
+  todosReducer,
+  ActionTodosAdd,
+  ActionTodosToggle,
+  ActionTodosFilter
+} from './todos.reducer';
+
+import { By } from '@angular/platform-browser';
 
 describe('TodosComponent', () => {
+  const resetStateMetaReducer = function(reducer) {
```

Comment:
This looks nice, what if we extracted this to test utils and made it generic so that in test we can pass any initial state we want so it can be used also for other reducers which don't have items and filter.

Then it could be used as `import { resetStateTestMetaReducer } from '../test-utils'`.
And it will be another wrapping function which accepts the initial state and returns `function(reducer)` like you implemented.

```javascript
function resetStateTestMetaReducer(resetTestState) {
   return function(reducer) {
      return function(state, action) {
          if (action.type === INIT) { // from import { INIT } from '@ngrx/store';
              state = resetTestState;
          }
          return reducer(state, action);
      }
   }
}
```

Then it can be used like `todos: resetStateMetaReducer({ items: [], filter: 'ALL' })(todosReducer)`.

I would put `test-utils.ts` next to `test.ts` in the `/src` folder.

What do you think?

Reply:
Grate point, only i must note that    INIT must be imported from '@ngrx/effects' (not from @ngrx/store)